silkrpc/main.cpp

```diff
@@ -14,6 +14,207 @@
+        void* tag;  // uniquely identifies a request.
+        bool ok;
+        while (true) {
+            // Block waiting to read the next event from the completion queue. The
+            // event is uniquely identified by its tag, which in this case is the
+            // memory address of a CallData instance.
+            // The return value of Next should always be checked. This return value
+            // tells us whether there is any kind of event or cq_ is shutting down.
+            GPR_ASSERT(cq_->Next(&tag, &ok));
+            GPR_ASSERT(ok);
+            static_cast<CallData*>(tag)->Proceed();
+        }
+    }
+
+    std::unique_ptr<ServerCompletionQueue> cq_;
+    EthService::AsyncService service_;
+    std::unique_ptr<Server> server_;
+};
+
+//--------------------------------------------------------
+int main(int argc, char** argv) {
+    ServerImpl server;
+    server.Run();
+
+    return 0;
+}
+
+const char* STR_FAKE_BLOCK =
+    "{\"author\": \"0x0193d941b50d91be6567c7ee1c0fe7af498b4137\",\n"
+    "\"difficulty\": \"0x3ffffa000\",\n"
+    "\"extraData\": \"0x476574682f76312e302e302f6c696e75782f676f312e342e32\",\n"
+    "\"gasLimit\": \"0x1388\",\n"
+    "\"gasUsed\": \"0x0\",\n"
+    "\"hash\": \"0xc63f666315fa1eae17e354fab532aeeecf549be93e358737d0648f50d57083a0\",\n"
+    "\"logsBloom\": "
+    "\"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+    "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+    "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+    "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+    "000000000000000000000000000000000000000000000000000000000000\",\n"
+    "\"miner\": \"0x0193d941b50d91be6567c7ee1c0fe7af498b4137\",\n"
+    "\"mixHash\": \"0xbe4ba21fe1ecb061e44f178428c772d2a0f59a7aafb5ed4e198eba4df3656e52\",\n"
+    "\"nonce\": \"0x5f6a5cc5c36e6627\",\n"
+    "\"number\": \"0xc\",\n"
+    "\"parentHash\": \"0x3f5e756c3efcb93099361b7ddd0dabfeaa592439437c1c836e443ccb81e93242\",\n"
+    "\"receiptsRoot\": \"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421\",\n"
+    "\"sealFields\": [\n"
+    "  \"0xa0be4ba21fe1ecb061e44f178428c772d2a0f59a7aafb5ed4e198eba4df3656e52\",\n"
+    "  \"0x885f6a5cc5c36e6627\"\n"
+    "],\n"
+    "\"sha3Uncles\": \"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347\",\n"
+    "\"size\": \"0x219\",\n"
+    "\"stateRoot\": \"0x821c41f30a2fd9580605363784a8a2a6575b255ec37cacf87fe52715b8828d8e\",\n"
+    "\"timestamp\": \"0x55ba42c0\",\n"
+    "\"totalDifficulty\": \"0x33f2ffe033\",\n"
+    "\"transactions\": [],\n"
+    "\"transactionsRoot\": \"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421\",\n"
+    "\"uncles\": []}";
+
+#endif
```

Comment:
I wouldn't put this server implementation related to the `eth.proto` interface in the `Silkrpc` main, please see below my comment for the `eth.proto` file.